File: src/main/java/com/diegolirio/revendaveiculos/service/VeiculoService.java
package com.diegolirio.revendaveiculos.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.diegolirio.revendaveiculos.dao.VeiculoDao;
import com.diegolirio.revendaveiculos.model.Veiculo;

@Service("veiculoService")
public class VeiculoService extends AbstractGenericService<Veiculo> {

	@Autowired
	private VeiculoDao veiculoDao;

	public List<Veiculo> getListHome() {
		return this.veiculoDao.getListHome();
	}

}

File: src/main/webapp/WEB-INF/static/core/js/service/VeiculoService.js
/**
 * VeiculoService
 */
app.factory('VeiculoService', ['$http', function($http) {
	
	var _getVeiculosHome = function() {
		return $http.get('/revendaveiculos/veiculo/get/list/home');
	};
	
	var _getPorId = function(id) {
		return $http.get('/revendaveiculos/veiculo/get/'+id);
	};
	
	var _getList = function() {
		return $http.get('/revendaveiculos/veiculo/get/list');
	};
	
	return {
		
		getVeiculosHome : _getVeiculosHome,
		
		getPorId : _getPorId,
		
		getList : _getList
		
	};
	
}]);